### ISSUE TITLE
Support for influxdb 0.9 [$215]

### ISSUE THREAD

**Issue by napalm272** [node-influx/node-influx#60]
Will node-influx support influxdb 0.9?  I think writePoint needs to support tags.

**Comment by bencevans** [comment]
`node-influx` doesn't have any backing for development other than the community that use it. Myself or one of the other maintainers may add this functionality if it interests them but we're very appreciative of pull requests to the project :)

**Comment by psvensson** [comment]
OK, I added $100 to the bounty. Pretty Please?

**Comment by nichdiekuh** [comment]
I'm looking into this, but it's a tricky and slow process. There are a lot of changes in 0.9 regarding the API. Some features have been removed or simplified (for instance: 'SHOW USERS' is not database specific anymore). 
It probably takes some time.

**Comment by psvensson** [comment]
OK, I hear you. Thanks for starting on this, though. Is there any specific tough nut to crack that you think one could help with, without having to read up on and understand both InfluxDB and the current client library? I have limited time myself but can chip in a bit if it's clearly defined.

**Comment by nichdiekuh** [comment]
All I can say now, is that there will be quite a lot of breaking changes. I haven't looked at the 0.9 release before (I'm still on 0.8) and I'm currently working through the changes function by function.

The way points/values are written (with and without tags) has changed completely. I'm probably also  going to add an automatic result-parser (which can be turned off per request) to merge the column names and values (tags) into a single array per point.

**Comment by bencevans** [comment]
Thanks Steffen! Not to worry about breaking changes, needed to happen again
at some point. 4.x here we come :D

**Comment by hampsterx** [comment]
hi there,

Can you update the readme to make it obvious this package is not (yet) compatible with 0.9x, I wasted about half an hour due to this :(

Thanks and great project~

**Comment by nichdiekuh** [comment]
Good point @hampsterx, I've updated the readme.

Now that I spent a day with 0.9.0 I remember why I postponed this task. It's quite unstable even though they released many RCs before the final version. For me, InfluxDB crashes frequently while quering data. In worst case you'll have to wait for a more stable release (0.9.1) before we can release an updated version of node-influx.

**Comment by mrenz** [comment]
Hi,

I just wanted to say that I started working on the support for version 0.9 about a week ago. I had to rewrite the client a lot but that was to be expected with the bunch of new features that were introduced with version 0.9.
I will make a pull request in the next couple of days. I already included most of the management features since they were easy to implement, next on my to-do list is the insertion of data and querying of data.

Greetings from Germany,
Manuel

**Comment by psvensson** [comment]
Wow, that's tough. I'll go pester them (nicely) a bit over there as well, then :)

**Comment by mrenz** [comment]
@nichdiekuh just wanted to let you know that I already rewrote the entire client for the support of 0.9 So we don't end up doing the same work twice. The missing features that I haven't implemented right now can be included very fast. I'll keep you up to date how it goes and I will make a pull request within the next days

**Comment by nichdiekuh** [comment]
@mrenz schick mir mal bitte 'ne kurze Mail an steffen ät nrg-media.de ;)

**Comment by fed135** [comment]
+1

**Comment by nichdiekuh** [comment]
Good news everyone!
With the new 0.9.1 RC1 release of InfluxDB and the help of @mrenz, an updated version of node-influx should be available soon. The code is almost feature-complete needs a bit more polish. The tests are finally working without any issues using 0.9.1-rc1.

In case anyone is interested in a development version, take a look at my fork: https://github.com/nichdiekuh/node-influx
I haven't had the time to update the docs yet, so please read the `tests.js`.

**Comment by psvensson** [comment]
Beautiful! Thanks both/all of you for doing this!

**Comment by mrenz** [comment]
Well, thank you @psvensson for backing this up!
Great to hear that the new version is available so soon. Also, it was great working with you @nichdiekuh. Let me know in case I can help you with the code!

**Comment by nichdiekuh** [comment]
@psvensson Wow, you're really generous! 
That's motivating and used this sunday morning to update the docs and the master branch. We are close to a release now.
Would be happy if you guys could clone and test the current masterbranch.

**Comment by psvensson** [comment]
Thanks :) I think it's an important project and since you now were two (at
least) working on it, it seemed the decent thing to do.
Will clone and check out tomorrow morning (CEST)

Cheers,
PS

On Sun, Jun 28, 2015 at 12:09 PM, Steffen <notifications@github.com> wrote:

> @psvensson <https://github.com/psvensson> Wow, you're really generous!
> That's motivating and used this sunday morning to update the docs and the
> master branch. We are close to a release now.
> Would be happy if you guys could clone and test the current masterbranch.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bencevans/node-influx/issues/60#issuecomment-116251930>
> .
>



-- 
"Easily the funniest and most original programming competition around";
http://www.escapedevnull.com

http://se.linkedin.com/in/petersvensson (LinkedIn)
https://plus.google.com/u/0/109826099655556298824
<https://plus.google.com/u/0/109826099655556298824/posts> (Google+)
http://twitter.com/psvensson (@psvensson)

**Comment by mrenz** [comment]
So I just made a clone and tested the current masterbranch.
On my Ubuntu VM with InfluxDb v0.9.1rc1 I got the result "61 passing (19s)" with mocha.

I also find the code structure very good and I like the option that one can get unparsed results with queryRaw for increased performance.

**Comment by psvensson** [comment]
Same here;
 ․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․

  61 passing (11s)

:)

Good stuff!



On Mon, Jun 29, 2015 at 10:18 AM, Manuel Renz <notifications@github.com>
wrote:

> So I just made a clone and tested the current masterbranch.
> On my Ubuntu VM with InfluxDb v0.9.1rc1 I got the result "61 passing
> (19s)" with mocha.
>
> I also find the code structure very good and I like the option that one
> can get unparsed results with queryRaw for increased performance.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/node-influx/node-influx/issues/60#issuecomment-116519094>
> .
>



-- 
"Easily the funniest and most original programming competition around";
http://www.escapedevnull.com

http://se.linkedin.com/in/petersvensson (LinkedIn)
https://plus.google.com/u/0/109826099655556298824
<https://plus.google.com/u/0/109826099655556298824/posts> (Google+)
http://twitter.com/psvensson (@psvensson)

**Comment by nichdiekuh** [comment]
I'm happy to hear the tests work for you, but I'm more worried about functional issues. Are the write/query functions useful the way they are? 
I've tried to add as many possibilities for writePoint/writePoints/writeSeries as possible and hope it's useful the way it is. 
Also, the query function merges the responses from all series that returned data and sorts the results by time. It seemed logical to me, but it might not be for others ;)

**Comment by nichdiekuh** [comment]
oh, I almost forgot. @bencevans suggest following the code style guidelines  by https://github.com/feross/standard and I think that's a good idea to keep everything clean. It's also easy to follow for all contributors.
The only thing bugging me is, that my beloved yoda expressions are considered as bad style ;)

**Comment by psvensson** [comment]
Sadly, I've never actually worked with influx before, since when I tried (a
couple of weeks ago) it wasn't really possible, which led us through the
events of last week with a valiant effort on the parts of the committers,
but I have too little experience at the moment to have any preference,
leaving that to those to have (of course).

On Mon, Jun 29, 2015 at 10:56 AM, Steffen <notifications@github.com> wrote:

> I'm happy to hear the tests work for you, but I'm more worried about
> functional issues. Are the write/query functions useful the way they are?
> I've tried to add as many possibilities for
> writePoint/writePoints/writeSeries as possible and hope it's useful the way
> it is.
> Also, the query function merges the responses from all series that
> returned data and sorts the results by time. It seemed logical to me, but
> it might not be for others ;)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/node-influx/node-influx/issues/60#issuecomment-116535428>
> .
>



-- 
"Easily the funniest and most original programming competition around";
http://www.escapedevnull.com

http://se.linkedin.com/in/petersvensson (LinkedIn)
https://plus.google.com/u/0/109826099655556298824
<https://plus.google.com/u/0/109826099655556298824/posts> (Google+)
http://twitter.com/psvensson (@psvensson)

**Comment by nichdiekuh** [comment]
@psvensson Lucky you :) You haven't been through the pain of migrating GBs of data from version to version with InfluxDB.

I'm going to wait until first of July for additional feedback. As long as nobody finds any serious issues with it, I'm going to push it to npm on wednesday/thursday.

**Comment by nichdiekuh** [comment]
Small update:
I'm still waiting for feedback from the InfluxDB guys, regarding an issue with SSL - I can't get it to work, neither on my server nor travis. If anyone reading this knows how to configure SSL properly, please leave  comment!

**Comment by mrenz** [comment]
@nichdiekuh what exactly is the problem with SSL? That the InfluxDB Server itself isn't running or is it a problem with our client? I can try it myself tomorrow.

**Comment by nichdiekuh** [comment]
With SSL configured InfluxDB won't start for me. Hence the HTTPS is commented out in the testsuite.

**Comment by mrenz** [comment]
@nichdiekuh I couldn't get it to work either. For me InfluxDB would still start but it simply ignores my ssl settings, no matter where I try to set them. I think we'll have to wait for someone from the influx team to reply.

**Comment by psvensson** [comment]
OK, standing by. Also, this is the first time I back something in this
manner, so please bump me if I need to do something at some point. Thx ^_^

On Thu, Jul 2, 2015 at 2:50 PM, Manuel Renz <notifications@github.com>
wrote:

> @nichdiekuh <https://github.com/nichdiekuh> I couldn't get it to work
> either. For me InfluxDB would still start but it simply ignores my ssl
> settings, no matter where I try to set them. I think we'll have to wait for
> someone from the influx team to reply.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/node-influx/node-influx/issues/60#issuecomment-118022415>
> .
>



-- 
"Easily the funniest and most original programming competition around";
http://www.escapedevnull.com

http://se.linkedin.com/in/petersvensson (LinkedIn)
https://plus.google.com/u/0/109826099655556298824
<https://plus.google.com/u/0/109826099655556298824/posts> (Google+)
http://twitter.com/psvensson (@psvensson)

**Comment by mrenz** [comment]
@psvensson we are in the same boat here, first time for me as well. But I'm sure we'll get it to work!


@nichdiekuh I tried to write points with the 0.9 version of the client but I am having problems:

- In the Readme it says client.writePoint(seriesName, values, tags, [options,] function(err, response) { }). However, when looking at the actual implementation, the code is InfluxDB.prototype.writePoint = function(seriesName, values, options, callback). Does the Readme need an update there?
--> What is the proper way of providing tags with .writePoint?

- How do I pass a JSON object as a value for a write? Whenever I try to send the object, I get the error response from the server 'Error: unable to parse xxx: invalid boolean'. The same thing happens if I try to provide it with JSON.stringify(). Do I have to escape the string manually beforehand?

**Comment by nichdiekuh** [comment]
I got a response regarding SSL, it's non-functional in 0.9, so we're good to go.

@mrenz 
There are indeed a few tiny mistakes in the documentation, but you can write single points like this:

´´´js
//write a single point with two values and two tags. time is omitted
client.writePoint(info.series.name, {value: 232, value2: 123}, { foo: 'bar', foobar: 'baz'}, done)

//write a single point with the value "1". The value "1" corresponds to { value : 1 }
client.writePoint(info.series.name, 1, { foo: 'bar', foobar: 'baz'}, done)
```

Are you sure you are loading the latest version from GIT?

**Comment by nichdiekuh** [comment]
Version 4.0.0 is now available via npm. I'm closing this issue now.

**Comment by nichdiekuh** [comment]
I've claimed the bounty and I'm gonna send @mrenz a fair share for this efforts :) Thank you Manuel for your contribution. And thanks to @psvensson  and @RobertHerhold for backing this.

**Comment by psvensson** [comment]
Sorry to be late, I was out on a dinner last night and couldn't accept by
mobile phone, apparently. Well done!

On Thu, Jul 2, 2015 at 8:08 PM, Steffen <notifications@github.com> wrote:

> I've claimed the bounty and I'm gonna send @mrenz
> <https://github.com/mrenz> a fair share for this efforts :) Thank you
> Manuel for your contribution. And thanks to @psvensson
> <https://github.com/psvensson> and @RobertHerhold
> <https://github.com/RobertHerhold> for backing this.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/node-influx/node-influx/issues/60#issuecomment-118113052>
> .
>



-- 
"Easily the funniest and most original programming competition around";
http://www.escapedevnull.com

http://se.linkedin.com/in/petersvensson (LinkedIn)
https://plus.google.com/u/0/109826099655556298824
<https://plus.google.com/u/0/109826099655556298824/posts> (Google+)
http://twitter.com/psvensson (@psvensson)
